### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by adelq**
A port of the project to Python 3. Fully compatible with Python 2.

~~Also, it needs to wait on https://github.com/dnet/pysecstr/issues/3.~~

### CODE REVIEW

**Comment by adelq** [comment]
@stef, this pull request is ready to go, and SecureString has been updated in PyPI to make this possible. As far as I have tested, it works flawlessly in Python 2.7, 3.4, 3.5, and 3.6, so they have been added to Travis CI.

One particular test (`test_sign_fail`) fails when the generated input has a null byte (which happens occasionally), it actually works though.

**Comment by stef** [comment]
thanks a lot!
can i ask you why you are doing this?
i have little time to review this the next week, will have to revisit this only after, i hope you don't mind.

**Comment by adelq** [comment]
No problem!

I found this project by coincidence, and the concept of a hackable, simpler PGP/GPG built on standard crypto libraries was really appealing to me.

No worries, take your time!

File: pbp/pbp.py [hunk 1]
```diff
@@ -1,38 +1,17 @@
...
+from .config import scrypt_salt
 
 ASYM_CIPHER = 5
 BLOCK_CIPHER = 23
 BLOCK_SIZE = 32*1024
 
-defaultbase='~/.pbp'
-scrypt_salt = 'qa~t](84z<1t<1oz:ik.@IRNyhG=8q(on9}4#!/_h#a7wqK{Nt$T?W>,mt8NqYq&6U<GB1$,<$j>,rSYI2GRDd:Bcm'
-
-def getkey(size, pwd='', empty=False, text=''):
```
Comment (stef) [inline pbp/pbp.py]: why did you move this to publickey?

File: pbp/publickey.py [hunk 2]
```diff
@@ -246,24 +254,44 @@ def buffered_verify(infd, outfd, basedir, self = None):
...
+
 def get_public_keys(basedir=None):
-    if not basedir: basedir=pbp.defaultbase
+    if not basedir: basedir=defaultbase
     basedir=os.path.expandvars(os.path.expanduser(basedir))
     pk_dir = get_pk_dir(basedir)
     if not os.path.exists(pk_dir):
         return
-    for root, ext in imap(os.path.splitext, os.listdir(pk_dir)):
+    for root, ext in map(os.path.splitext, os.listdir(pk_dir)):
```
Comment (stef) [inline pbp/publickey.py]: i prefer an iterator here

File: pbp/publickey.py [hunk 3]
```diff
@@ -246,24 +254,44 @@ def buffered_verify(infd, outfd, basedir, self = None):
...
 def get_secret_keys(basedir=None):
-    if not basedir: basedir=pbp.defaultbase
+    if not basedir: basedir=defaultbase
     basedir=os.path.expandvars(os.path.expanduser(basedir))
     seen = set()
     sk_dir = get_sk_dir(basedir)
     if not os.path.exists(sk_dir):
         return
-    for root, ext in imap(os.path.splitext, os.listdir(sk_dir)):
+    for root, ext in map(os.path.splitext, os.listdir(sk_dir)):
```
Comment (stef) [inline pbp/publickey.py]: an iterator would be preferable here as well.

File: pbp/publickey.py (hunk 2 above)
Comment (adelq) [reply pbp/publickey.py]: This is an iterator still in both Python 2 and Python 3, note the import at the top of the file. `imap` is still imported but renamed to `map`, and `map` is an iterator by default in Python 3.

File: pbp/publickey.py (hunk 3 above)
Comment (adelq) [reply pbp/publickey.py]: See explanation above. This is still an iterator.

File: pbp/pbp.py (hunk 1 above)
Comment (adelq) [reply pbp/pbp.py]: To avoid circular imports, `getkey` is used by `publickey.py` and `pbp.py` imports `publickey.py`. This is not a problem in Python 2 because of relative imports. In Python 3.4, an error is raised for circular imports since it requires explicit relative imports but also does not support resolving circular dependencies (as Python 3.5+ does).

Comment (adelq) [reply pbp/pbp.py]: Circular imports is also why `defaultbase` and `scrypt_salt` were moved to `config.py`, I tried to keep refactoring to a minimum, but was needed to resolve circular imports.

**Comment by adelq** [comment]
thanks for taking a look, I answered in the comments

**Comment by stef** [comment]
thanks for the hard work! great job. hope you enjoy pbp ;)

**Comment by adelq** [comment]
thanks!
